lib/ansible/modules/network/onyx/onyx_syslog_files.py

```diff
@@ -0,0 +1,214 @@
+        elif max_num and (int(max_num) < 0 or int(max_num) > self.MAX_FILES):
+            self._module.fail_json(
+                msg='logging max_num must positive number less than {0}'.format(self.MAX_FILES))
+
+    def validate_upload_url(self, upload_url):
+        check = self.URL_REGEX.match(upload_url)
+        if upload_url and not check:
+            self._module.fail_json(
+                msg='Invalid url, make sure that you use "[ftp, scp, tftp, sftp]://username:password@hostname:/location" format')
+
+    def validate_rotation_max_num(self, max_num):
+        if max_num and (max_num < 1 or max_num > self.MAX_FILES):
+            self._module.fail_json(
+                msg='logging max_num must be between 1 and 999999')
+
+    def show_logging(self):
+        return show_cmd(self._module, "show logging", json_fmt=True, fail_on_error=False)
+
+    def load_current_config(self):
+        self._current_config = dict()
+        current_config = self.show_logging()[0]
+        if 'Log rotation frequency' in current_config:
+            freq = current_config.get('Log rotation frequency')  # daily (Once per day at midnight)
+            freq_str = freq.split()[0]
+            self._current_config['frequancy'] = freq_str
+
+        if 'Log rotation (debug) size threshold' in current_config:  # 19.07 megabytes
+            size = current_config.get('Log rotation (debug) size threshold')
+            size_str = size.split()[0]
+            self._current_config['size'] = float(size_str)
+
+        if 'Log rotation size threshold' in current_config:  # 10.000% of partition (987.84 megabytes)
+            size = current_config.get('Log rotation size threshold')
+            size_str = size.split()[0].replace('%', '')
+            self._current_config['size_pct'] = float(size_str)
+
+        if 'Number of archived log files to keep' in current_config:
+            max_num = current_config.get('Number of archived log files to keep')
+            self._current_config['max_num'] = int(max_num)
+
+    def get_required_config(self):
+        self._required_config = dict()
+        required_config = dict()
+        module_params = self._module.params
+
+        if module_params.get('delete_group', None):
+            required_config['delete_group'] = module_params['delete_group']
+        if module_params.get('upload_file', None):
+            required_config.update({'upload_file': module_params['upload_file'],
+                                    'upload_url': module_params['upload_url']})
+        if module_params.get('rotation', None):
+            required_config['rotation'] = module_params['rotation']
+
+        self.validate_param_values(required_config)
+        self._required_config = required_config
+
+    def generate_commands(self):
+        required_config = self._required_config
+        current_config = self._current_config
+        if required_config.get('rotation', None):
+            rotation = required_config['rotation']
```

Comment:
multiple dictionary access:
instead of
```python
if required_config.get('rotation', None):
    rotation = required_config['rotation']
```
use:
```python
rotation = required_config('rotation')
if rotation is not None:
    # do your stuff
```

use this rule for all dictionary access below